### PR TITLE
[SPARK-27276][PYTHON][DOCS][FOLLOW-UP] Update documentation about Arrow version in PySpark as well

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

Looks updating documentation from 0.8.0 to 0.12.1 was missed.

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @BryanCutler

File: docs/sql-pyspark-pandas-with-arrow.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ license: |
 * Table of contents
 {:toc}
 
-## Apache Arrow in Spark
+## Apache Arrow in PySpark
```
Comment (HyukjinKwon) [inline docs/sql-pyspark-pandas-with-arrow.md]: I realised that other Arrow optimization might likely be placed in other places .. :).

Comment (viirya) [reply docs/sql-pyspark-pandas-with-arrow.md]: Make sense. The filename is sql-pyspark...

**Review by viirya — APPROVED** [review]
This looks good.

There is another section:

Supported SQL Types
Currently, all Spark SQL data types are supported by Arrow-based conversion except MapType, ArrayType of TimestampType, and nested StructType. BinaryType is supported only when installed PyArrow is equal to or higher than 0.10.0.

As currently supported version is 0.12.1, the last sentence looks redundant?

**Comment by HyukjinKwon** [comment]
Yea, thanks for checking it.

**Review by srowen — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]

**Review by BryanCutler — APPROVED** [review]
LGTM

**Comment by BryanCutler** [comment]
merged to master, thanks @HyukjinKwon !

**Pull request closed without merging**